lib/model/CEventRatePopulationModel.cc

```diff
@@ -459,12 +461,26 @@ void CEventRatePopulationModel::sample(core_t::TTime startTime,
 
             for (auto& attribute : attributeValuesAndWeights) {
                 std::size_t cid = attribute.first;
+
+                const auto createAndAddAnnotation =
```

Comment:
I think we should clear the `m_CurrentBucketStats.s_Annotations` at the start of this function. It'll get called exactly once per bucket and we'll write them out between calls.